slowmovie.py

```diff
@@ -10,205 +10,349 @@
+from waveshare_epd import epd7in5_V2 as epd_driver
+
+def clamp(n, smallest, largest):
+    return max(smallest, min(n, largest))
+
+# Handle when the program is killed and exit gracefully
+def signal_handler(signum, frame):
+    print('\nExiting Program')
+
+    # Call init to get gpio pins setup correctly and then exit properly
+    epd_driver.epdconfig.module_init()
+    epd_driver.epdconfig.module_exit()
+    exit(0)
+
+def generate_frame(in_filename, out_filename, time):
+    try:
+        (
+            ffmpeg
+            .input(in_filename, ss=time)
+            .filter('scale', 'iw*sar', 'ih')
+            .filter('scale', width, height, force_original_aspect_ratio=1)
+            .filter('pad', width, height, -1, -1)
+            .output(out_filename, vframes=1)
+            .overwrite_output()
+            .run(capture_stdout=True, capture_stderr=True)
+        )
+    except ffmpeg.Error as e:
+        print('stdout:', e.stdout.decode('utf8'))
+        print('stderr:', e.stderr.decode('utf8'))
+        raise e
+
+def get_video_info(filepath):
+    probeInfo = ffmpeg.probe(filepath)
+
+    stream = probeInfo['streams'][0]
+
+    # Calculate framerate
+    # https://github.com/fepegar/utils/commit/f52ab5152eaa5b8ceadefee172e985aaab3a9947#diff-521f42b0315f6bf8900be8407965552a8f6b32d9a22c09178c55be62b1bef4a2R23-R39
+    r_fps = stream['r_frame_rate']
+    try:
+        num, denom = r_fps.split('/')
+    except ValueError:
+        # If r_fps isn't a fraction (does this happen?)
+        num = r_fps
+        denom = 1
+    fps = float(num) / float(denom)
+
+    # Calculate duration
+    duration = float(probeInfo['format']['duration'])
+
+    # Either get frame count or calculate it
+    try:
+        # Get frame count for .mp4s
+        frameCount = int(stream['nb_frames'])
+    except KeyError:
+        # Calculate frame count for .mkvs (and maybe other formats?)
+        frameCount = int(duration * fps)
+
+    # Calculate frametime (ms each frame is displayed)
+    frameTime = 1000 / fps
+
+    return {
+        'frame_count' : frameCount,
+        'fps' : fps,
+        'duration' : duration,
+        'frame_time' : frameTime }
+
+def is_vid(file):
+    name, ext = os.path.splitext(file)
+    return ext.lower() in fileTypes
+
+def check_vid(file):
+    if not is_vid(file):
+        raise argparse.ArgumentTypeError(f"File '{file}' should be a file with one of the following supported extensions: {', '.join(fileTypes)}")
+    return file
+
+def check_dir(dir):
+    if os.path.isdir(dir):
+        return dir
+    # Make default videos directory if it doesn't exist and it's needed
+    elif dir == 'Videos':
+        os.mkdir(dir)
+    else:
+        raise argparse.ArgumentTypeError(f"Directory '{dir}' could not be found")
+
+# Calculate how long it'll take to play a video.
+# output value: 'd[ay]', 'h[our]', 'm[inute]', 's[econd]'
+def estimate_runtime(delay, increment, videoLengthS, videoFPS, output):
```

Comment:
Could a default for the ```output``` parameter be something like a "guess" for the best output? I think this is better than hardcoding "days" as for the call below. Check if  60 > s, check min, if 60 > min, show hours, etc.

Reply:
I considered this but was lazy. This is the push I needed. Added in 42121a9